maestro-ios-driver/src/main/kotlin/xcuitest/XCTestDriverClient.kt

```diff
@@ -197,16 +197,22 @@ class XCTestDriverClient(
         return okHttpClient.newCall(requestBuilder.build()).execute()
     }
 
-    private fun OkHttpClient.Builder.addRetryOnErrorInterceptor() = addInterceptor(Interceptor {
-        val request = it.request()
-        try {
-            it.proceed(request)
-        } catch (connectException: IOException) {
-            installer.start()?.let { newClient ->
-                client = newClient
-                it.proceed(request)
-            } ?: throw XCTestDriverUnreachable("Failed to reach out XCUITest Server in RetryOnError")
+    private fun OkHttpClient.Builder.addRetryInterceptor() = addInterceptor(Interceptor { chain ->
+        var exception: IOException? = null
+        repeat(3) {
+            try {
+                val response = chain.proceed(chain.request())
+                if (response.isSuccessful) {
+                    return@Interceptor response
+                }
+            } catch (e: IOException) {
+                exception = e
+            }
+            Thread.sleep(200L)
```

Comment:
Confused about this sleep 🤔 ?

Reply:
Just to wait 200ms between attempts. Instead of retrying right away. Avoiding overloading the XCTest Server.

Reply:
Got it and when do we see this retry happening I see its done in IOExceptions are there special cases we want to handle. I'm asking this because retryConnectionOnFailure(true) do cover most of it. So, either we have some specific case which is not covered by retryConnectionOnFailure 🤔